celery/backends/redis.py

```diff
@@ -102,9 +103,11 @@ def on_state_change(self, meta, message):
         self._maybe_cancel_ready_task(meta)
 
     def start(self, initial_task_id, **kwargs):
-        self._pubsub = self.backend.client.pubsub(
-            ignore_subscribe_messages=True,
-        )
+        if self._pubsub is None:
+            self._pubsub = self.backend._create_client(
```

Comment:
I think the usage of a (new) public method which is the non-cached equivalent of `RedisBackend.client` (see [here](https://github.com/celery/celery/pull/5145/files#diff-accb7090ee45fffb2262cbceeabef002R419)) would require much less testing and simplify this section.